### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by shafran-andriy**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,47 @@
...
+    def __init__(self, distance_from_city_center: float = 0,
+                 clean_power: int = 0,
+                 average_rating: float = 0,
+                 count_of_ratings: int = 0) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = round(average_rating, 1)
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, list_of_cars: list) -> float:
```
Comment (EdAlekseiev) [inline app/main.py]: Specify inner type of list
```suggestion
    def serve_cars(self, list_of_cars: list[Car]) -> float:
```

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (shafran-andriy) [reply app/main.py]: Fixed

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,47 @@
...
+    def __init__(self, distance_from_city_center: float = 0,
+                 clean_power: int = 0,
+                 average_rating: float = 0,
+                 count_of_ratings: int = 0) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = round(average_rating, 1)
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, list_of_cars: list[Car]) -> float:
```
Comment (vasylhnatiuk) [inline app/main.py]:
```suggestion
    def serve_cars(self, cars: list[Car]) -> float:
```

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,47 @@
...
+        income = 0
+        for car in list_of_cars:
+            if car.clean_mark < self.clean_power and self.wash_single_car(car):
+                income += self.calculate_washing_price(car)
+                car.clean_mark = self.clean_power
+        return round(income, 1)
+
+    def calculate_washing_price(self, attrs_car: Car) -> float:
```
Comment (vasylhnatiuk) [inline app/main.py]:
```suggestion
    def calculate_washing_price(self, car: Car) -> float:
```

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,47 @@
...
+        comfort_class = attrs_car.comfort_class
+        clean_mark = attrs_car.clean_mark
+        price = (comfort_class
+                 * ((self.clean_power - clean_mark)
+                    * self.average_rating) / self.distance_from_city_center)
+        return price
+
+    def wash_single_car(self, attrs_car: Car) -> Car:
```
Comment (vasylhnatiuk) [inline app/main.py]:
```suggestion
    def wash_single_car(self, car: Car) -> Car:
```

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (vasylhnatiuk) [reply app/main.py]: use simple names

File: app/main.py (hunk 3 above)
Comment (shafran-andriy) [reply app/main.py]: fixed

File: app/main.py (hunk 4 above)
Comment (shafran-andriy) [reply app/main.py]: Also fixed

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**